core/src/test/java/org/apache/calcite/test/SqlValidatorTest.java

```diff
@@ -11685,6 +11685,19 @@ private void checkCustomColumnResolving(String table) {
             + "MYFUN\\(<NUMERIC>, <NUMERIC>\\).*");
   }
 
+  @Test void testPositionalAggregateWithExpandedCurrentDateFunction() {
+    SqlConformance defaultPlusOrdinalGroupBy =
+        new SqlDelegatingConformance(SqlConformanceEnum.DEFAULT) {
+          @Override public boolean isGroupByOrdinal() {
+            return true;
+          }
+        };
+    sql("SELECT HIREDATE >= CURRENT_DATE, COUNT(*) "
+        + "FROM EMP GROUP BY 1")
```

Comment:
I'm confused by why the `current_date` caused the problem, does other non-param function triggers the bug too ?

Reply:
Please take a look at the stack trace from [CALCITE-4207](https://issues.apache.org/jira/browse/CALCITE-4207) description and [this comment](https://issues.apache.org/jira/browse/CALCITE-4207?focusedCommentId=17189063&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17189063).

Reply:
> In one place CURRENT_DATE was represented as SqlBasicCall while in other it was instance of SqlIdentifier

Still does not got your point, in you fix, you just use the expanded group expr for validation, but the `CURRENT_DATE` is not a group expr ?

Reply:
Positional aggregate points to the first expression in select. Validation tries to match 2 expression trees : 
1. The first item from select which is already expanded at the moment and contains current_date as an instance of SqlBasicCall 
2. An item from the group by where prior to my fix current_date was presented as an instance of the SqlIdentifier.

Due to this instance mismatch validation goes further and tells that `Expression 'EMP.HIREDATE' is not being grouped`, although grouping was performed by the whole expression `HIREDATE >= CURRENT_DATE`.

Reply:
Hello @danny0405 , could you proceed with reviewing the PR?

Reply:
I tried the fix locally and think the fix is reasonable.

Reply:
Thanks for review:)